authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,1006 @@
+    } else {
+      policy = new RangerPolicy();
+      policy.setService(rangerServiceName);
+      policy.setName(metadataObject.fullName());
+      policy.setPolicyLabels(Lists.newArrayList(MANAGED_BY_GRAVITINO));
+
+      List<String> nsMetadataObject =
+          Lists.newArrayList(SecurableObjects.DOT_SPLITTER.splitToList(metadataObject.fullName()));
+      if (nsMetadataObject.size() > 4) {
+        // The max level of the securable object is `catalog.db.table.column`
+        throw new RuntimeException("The securable object than 4");
+      }
+      nsMetadataObject.remove(0); // remove `catalog`
+
+      for (int i = 0; i < nsMetadataObject.size(); i++) {
+        RangerPolicy.RangerPolicyResource policyResource =
+            new RangerPolicy.RangerPolicyResource(nsMetadataObject.get(i));
+        policy
+            .getResources()
+            .put(
+                i == 0
+                    ? RangerDefines.RESOURCE_DATABASE
+                    : i == 1 ? RangerDefines.RESOURCE_TABLE : RangerDefines.RESOURCE_COLUMN,
+                policyResource);
+      }
+
+      RangerPolicy finalPolicy = policy;
+      ownerPrivileges.stream()
+          .forEach(
+              ownerPrivilege -> {
+                // Each owner's privilege will create one RangerPolicyItemAccess in the policy
+                RangerPolicy.RangerPolicyItem policyItem = new RangerPolicy.RangerPolicyItem();
+                policyItem
+                    .getAccesses()
+                    .add(new RangerPolicy.RangerPolicyItemAccess(ownerPrivilege));
+                if (newOwner != null) {
+                  if (newOwner.type() == Owner.Type.USER) {
+                    policyItem.getUsers().add(newOwner.name());
+                  } else {
+                    policyItem.getGroups().add(newOwner.name());
+                  }
+                }
+                finalPolicy.getPolicyItems().add(policyItem);
+              });
+    }
+    try {
+      if (policy.getId() == null) {
+        rangerClient.createPolicy(policy);
+      } else {
+        rangerClient.updatePolicy(policy.getId(), policy);
+      }
+    } catch (RangerServiceException e) {
+      throw new RuntimeException(e);
+    }
+
+    return Boolean.TRUE;
+  }
+
+  /**
+   * Because one Ranger policy maybe contain multiple Gravitino securable objects, <br>
+   * So we need to find the corresponding policy item mapping to set the user.
+   */
+  @Override
+  public Boolean onGrantedRolesToUser(List<Role> roles, User user) throws RuntimeException {
+    // If the user does not exist, then create it.
+    onUserAdded(user);
+
+    roles.stream()
+        .forEach(
+            role -> {
+              createRangerRoleIfNotExists(role.name());
+              GrantRevokeRoleRequest grantRevokeRoleRequest =
+                  createGrantRevokeRoleRequest(role.name(), user.name(), null);
+              try {
+                rangerClient.grantRole(rangerServiceName, grantRevokeRoleRequest);
```

Comment:
Will this client fail to grant role?

Reply:
Currently, we use admin to execute grant to role.
Just like E2E IT test case.

Reply:
Will it fail because of network timeout?

Reply:
@xunliu Could you reply this?

Reply:
I create a issue #4620 to track this problem